src/types/iterator.rs

```diff
@@ -248,4 +248,49 @@ def fibonacci(target):
             assert_eq!(iter_ref.get_refcnt(), 2);
         })
     }
+
+    #[test]
+    #[cfg(not(all(not(Py_3_8), Py_LIMITED_API)))]
+    #[cfg(feature = "macros")]
+    fn python_class_not_iterator() {
```

Comment:
might also be woth testing some of the other failure cases turned up in https://github.com/pydantic/pydantic-core/pull/372 (search for `xfail`):
* subclass of `str`
* subclass of `bytes`
* a plain class like:

```py
    class Foobar:
        def __repr__(self):
            return '<Foobar repr>'
```

Reply:
I'm confident enough in the underlying mechanism (really it's just checking the presence of a `__next__` implementation) that I think the one test is enough. What I _will_ add, though, is the opposite check - that one which has a `__next__` will pass the check.